core/src/main/scala/io/snappydata/TableStatsProviderService.scala

```diff
@@ -0,0 +1,293 @@
+package io.snappydata
+
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+
+import com.gemstone.gemfire.CancelException
+import com.gemstone.gemfire.cache.DataPolicy
+import com.gemstone.gemfire.cache.execute.FunctionService
+import com.gemstone.gemfire.i18n.LogWriterI18n
+import com.gemstone.gemfire.internal.SystemTimer
+import com.gemstone.gemfire.internal.cache.execute.InternalRegionFunctionContext
+import com.gemstone.gemfire.internal.cache.{PartitionedRegion, LocalRegion}
+import com.pivotal.gemfirexd.internal.engine.Misc
+import com.pivotal.gemfirexd.internal.engine.distributed.{GfxdMessage, GfxdListResultCollector}
+import com.pivotal.gemfirexd.internal.engine.distributed.GfxdListResultCollector.ListResultCollectorValue
+import com.pivotal.gemfirexd.internal.engine.sql.execute.MemberStatisticsMessage
+import com.pivotal.gemfirexd.internal.engine.store.{CompactCompositeKey, GemFireContainer}
+import com.pivotal.gemfirexd.internal.engine.ui.{SnappyRegionStatsCollectorFunction, SnappyRegionStatsCollectorResult, SnappyIndexStats, SnappyRegionStats}
+import com.pivotal.gemfirexd.internal.iapi.types.RowLocation
+import io.snappydata.Constant._
+
+import org.apache.spark.sql.catalyst.expressions.UnsafeRow
+import org.apache.spark.sql.collection.Utils
+import org.apache.spark.sql.execution.columnar.{JDBCSourceAsStore, JDBCAppendableRelation}
+import org.apache.spark.sql.execution.columnar.encoding.ColumnStatsSchema
+import org.apache.spark.unsafe.Platform
+import org.apache.spark.{SparkContext, Logging}
+import org.apache.spark.sql.{SnappySession, SnappyContext}
+
+trait TableStatsProviderService extends Logging {
+
+  @volatile
+  private var tableSizeInfo = Map.empty[String, SnappyRegionStats]
+  private val membersInfo = TrieMap.empty[String, mutable.Map[String, Any]]
```

Comment:
tableSizeInfo is volatile while membersInfo is not. Is this what is intended here?